### ISSUE TITLE
Acorn stops probing dependencies once encountering a linked dependency

### ISSUE THREAD

**Issue by mocdaniel** [acorn-io/acorn#1191]
Imagine the following minimal `Acornfile`

```cue
containers: {
    frontend: {
        image: "nginx"
        dependsOn: ["postgres", "redis"]  // SCENARIO 1
        dependsOn: ["redis", "postgres"]  // SCENARIO 2
    }

    postgres: {
        image: "postgres"
    }

    redis: {
        image: "redis"
    }
}
```

If I run this Acorn now with the following command 

`acorn run --link ext-postgres:postgres .`

I observe the following behaviour:

- `SCENARIO 1` spins up `frontend` immediately **without** probing `redis`, because Acorn apparently stops parsing dependencies after encountering an externally linked dependency
- `SCENARIO 2` keeps `frontend` pending until `redis` comes up, because `redis` precedes `postgres` within the `dependsOn` list 

Please see also [the corresponding thread on Slack](https://acorn-users.slack.com/archives/C03R9ME0SKC/p1675423188632459)

**Comment by sangee2004** [comment]
Able to reproduce the issue with following steps:

Had Acornfile with following container dependencies:

```
containers: {
  app: {
    dependsOn: ["test1","test2"]
    // other container params
  }
     test1:
{
  // other container params
}
 test2:
{
  // other container params
  Had readiness probe that would always fail
} 

```
Launch this Acorn following command :

`acorn run --link extapp:test1 .`

This resulted in `app` being created even when `test2` containers were is not ready state.

```
 % acorn apps
NAME        IMAGE               HEALTHY   UP-TO-DATE   CREATED     ENDPOINTS                                                                                                                                                                        MESSAGE
snowy-sky   84d74a3d599f        1/2       2            39s ago     http://app-snowy-sky-72c99328.local.on-acorn.io => app:80, http://test2-snowy-sky-15a4fb4e.local.on-acorn.io => test2:80                                                         [containers: test2 is not ready]

 % acorn containers 
NAME                               APP         IMAGE     STATE                 RESTARTCOUNT   CREATED     MESSAGE
snowy-sky.app-6d76668c99-fjw9n     snowy-sky   nginx     running               0              92s ago     
snowy-sky.test2-79d87b76cb-wtsmn   snowy-sky   nginx     running (not ready)   0              92s ago     
e
```


With latest acorn build having the fixes - `v0.5.1-32-g4398ca3+4398ca39` , we see the `app` is in `pending create` state as expected.

```
% acorn apps      
NAME        IMAGE          HEALTHY   UP-TO-DATE   CREATED     ENDPOINTS                                                                                                                  MESSAGE
aged-rain   20932737d06f   0/1       1            2m33s ago   http://app-aged-rain-0fa5fc24.local.on-acorn.io => app:80, http://test2-aged-rain-6cfce888.local.on-acorn.io => test2:80   [containers: app pending create; test2 is not ready]

% acorn containers
NAME                               APP         IMAGE     STATE                 RESTARTCOUNT   CREATED   MESSAGE
aged-rain.test2-5ffbd65ff9-5nv62   aged-rain   nginx     running (not ready)   0              17m ago 
```

**Comment by sangee2004** [comment]
@mocdaniel This issue is fixed on latest main builds - https://docs.acorn.io/installation/installing#development-binaries-main-build and will be made available in future release. Please feel free to reopen this issue if you still encounter this issue with builds having the fix.
